From: =?UTF-8?q?Mika=20Pfl=C3=BCger?= <debian@mikapflueger.de>
Date: Sun, 4 Mar 2012 02:28:12 +0100
Subject: authlogin policy adjustments: * Label /etc/.group.edit.swp and
 simliar files correctly * Also dontaudit open of shadow files by
 chkpwd

---
 policy/modules/system/authlogin.fc |    7 ++++---
 policy/modules/system/authlogin.if |    2 +-
 2 files changed, 5 insertions(+), 4 deletions(-)

diff --git a/policy/modules/system/authlogin.fc b/policy/modules/system/authlogin.fc
index 28ad538..29c6333 100644
--- a/policy/modules/system/authlogin.fc
+++ b/policy/modules/system/authlogin.fc
@@ -1,11 +1,12 @@
 
 /bin/login		--	gen_context(system_u:object_r:login_exec_t,s0)
 
-/etc/\.pwd\.lock	--	gen_context(system_u:object_r:shadow_t,s0)
-/etc/group\.lock	--	gen_context(system_u:object_r:shadow_t,s0)
 /etc/gshadow.*		--	gen_context(system_u:object_r:shadow_t,s0)
-/etc/passwd\.lock	--	gen_context(system_u:object_r:shadow_t,s0)
+/etc/\.group\.edit\.swp	--	gen_context(system_u:object_r:shadow_t,s0)
+/etc/\.gshadow\.edit\.swp --	gen_context(system_u:object_r:shadow_t,s0)
 /etc/shadow.*		--	gen_context(system_u:object_r:shadow_t,s0)
+/etc/\.passwd\.edit\.swp --	gen_context(system_u:object_r:shadow_t,s0)
+/etc/\.shadow\.edit\.swp --	gen_context(system_u:object_r:shadow_t,s0)
 
 /sbin/pam_console_apply	 --	gen_context(system_u:object_r:pam_console_exec_t,s0)
 /sbin/pam_timestamp_check --	gen_context(system_u:object_r:pam_exec_t,s0)
diff --git a/policy/modules/system/authlogin.if b/policy/modules/system/authlogin.if
index d51181f..f9cecd4 100644
--- a/policy/modules/system/authlogin.if
+++ b/policy/modules/system/authlogin.if
@@ -426,7 +426,7 @@ interface(`auth_domtrans_chkpwd',`
 
 	corecmd_search_bin($1)
 	domtrans_pattern($1, chkpwd_exec_t, chkpwd_t)
-	dontaudit $1 shadow_t:file { getattr read };
+	dontaudit $1 shadow_t:file { open getattr read };
 	auth_domtrans_upd_passwd($1)
 ')
 
